sync/class.jetpack-sync-listener.php

```diff
@@ -236,26 +236,23 @@ function enqueue_action( $current_filter, $args, $queue ) {
 	}
 
 	function get_actor() {
-		$current_user = wp_get_current_user();
-
-		$actor = array();
-		if ( $current_user ) {
-			$actor['display_name'] = $current_user->display_name;
-			$actor['user_email'] = $current_user->user_email;
-			$actor['user_roles'] = $current_user->roles; /* Since 5.0.0 */
-			$actor['translated_role'] = Jetpack::translate_current_user_to_role(); /* Since 5.0.0 */
-		}
-
-		if ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
-			$actor[ 'ip' ] = $_SERVER['REMOTE_ADDR'];
-		}
-
-		$actor['is_cron'] = defined( 'DOING_CRON' ) ? DOING_CRON : false;
-		$actor['is_wp_admin'] = is_admin();
-		$actor['is_rest'] = defined( 'REST_API_REQUEST' ) ? REST_API_REQUEST : false;
-		$actor['is_xmlrpc'] = defined( 'XMLRPC_REQUEST' ) ? XMLRPC_REQUEST : false;
-		$actor['is_wp_rest'] = defined( 'REST_REQUEST' ) ? REST_REQUEST : false;
-		$actor['is_ajax'] = defined( 'DOING_AJAX' ) ? DOING_AJAX : false;
+		$user = wp_get_current_user();
+
+		$actor = array(
+			'wpcom_user_id'    => null,
+			'external_user_id' => $user ? $user->ID : null,
+			'display_name'     => $user ? $user->display_name : null,
+			'user_email'       => $user ? $user->user_email : null,
+			'user_roles'       => $user ? $user->roles : null,
```

Comment:
Check instead for each property here. For example, `$user` might be set, but not `user_email`, so check `isset( $user->user_email )`.
WP core for example checks user ID like this in https://github.com/WordPress/WordPress/blob/master/wp-includes/user.php#L429